### DISCUSSION TITLE
How to load namespaces from node_modules?

### DISCUSSION THREAD

**Discussion by coderjib** [vinissimus/next-translate#572] [Q&A] [answered]
I am trying to import json namespaces from 2 modules (project dependencies), but they do not load.
What I want to do is the following: if the namespace starts with "c-" load namespaces from module-A, if it starts with "s-" load from module-B, otherwise load from local './locales' folder.
So I am using a custom loadLocaleFrom function to dynamically select the correct folder to source from based on namespace (it uses `path` and `require.resolve`).
But even if I have checked it calculates the correct path to the file (so there should be no error in the function logic), it does not load the translation. While if I manually hardcode the same json path returned from the function, it works. Why? Is it mandatory to hardcode a path directly in i18n.js file? Is there any recommended way or example of doing something similar to my requirement? Please note I am using v. 1.0.5.
Thanks a lot

**Comment by aralroca** [comment, marked as answer]
I think this has to do with how Webpack works to resolve dynamic imports. It needs to know in advance the path to create the necessary chunks. If the path is dynamic, Webpack does not know which chunks to create, and I guess the problem comes from here. 

Something like this:

```js
function loadLocaleFrom(lang, ns) {
 const path = someCondition ? './some-path' : '/another-path'
 return import(`{path}/${lang}/${ns}.json`).then((m) => m.default),
}
```

Webpack don't know the `path` in advance. So maybe it helps to change it to something like:

```js
function loadLocaleFrom(lang, ns) {
 if(someCondition) return  import(`./some-path/${lang}/${ns}.json`).then((m) => m.default)
 return  import(`./another-path/${lang}/${ns}.json`).then((m) => m.default)
}
```

This way, Webpack knows that should create chunks to all the content of `./some-path` and `./another-path`.

Hope it helps!
